pkg/labels/matcher.go

```diff
@@ -86,3 +89,70 @@ func (m *Matcher) Matches(s string) bool {
 	}
 	panic("labels.Matcher.Matches: invalid match type")
 }
+
+type Matchers []*Matcher
+
+func (ms Matchers) Len() int      { return len(ms) }
+func (ms Matchers) Swap(i, j int) { ms[i], ms[j] = ms[j], ms[i] }
+
+func (ms Matchers) Less(i, j int) bool {
+	if ms[i].Name > ms[j].Name {
+		return false
+	}
+	if ms[i].Name < ms[j].Name {
+		return true
+	}
+	if ms[i].Value > ms[j].Value {
+		return false
+	}
+	return ms[i].Value < ms[j].Value
+}
+
+// Matches checks whether all matchers are fulfilled against the given label set.
+func (ms Matchers) Matches(lset model.LabelSet) bool {
+	for _, m := range ms {
+		if !m.Matches(string(lset[model.LabelName(m.Name)])) {
+			return false
+		}
+	}
+	return true
+}
+
+func (ms Matchers) String() string {
+	var buf bytes.Buffer
+
+	buf.WriteByte('{')
+	for i, m := range ms {
+		if i > 0 {
+			buf.WriteByte(',')
+		}
+		buf.WriteString(m.String())
+	}
+	buf.WriteByte('}')
+
+	return buf.String()
+}
+
+// Match checks whether the label of the matcher has the specified
+// matching value.
+func (m *Matcher) Match(lset model.LabelSet) bool {
```

Comment:
This is only used in line 153 below, but it gives `Matcher` a method `Match(model.labelSet)` in addition to `Matches(string)`. That's a bit confusing. At least the naming should be fixed, perhaps to `MatchesLabelSet` or something`, and the method should be moved up in the file to the other methods of `Matcher`.

But perhaps, for now, it's better to inline this code into the method below (line 153, will suggest something there) and not have this method until it's needed elsewhere.

A also think this method is incorrect for negative matching. And the tests should actually catch that bug. Could you investigate why not?

Reply:
I have found out now: The correct method is already above, with the name `Matches`.

Reply:
So just delete this method entirely, together with the Matchers.Match method below.

Reply:
The code in Alertmanager itself uses the `Matches` method already. There is a reference to `Match` in the CLI code. This simply has to be changed to `Matches`, see below.